### PR TITLE
feat(cargo-shuttle): check project name available

### PR DESCRIPTION
**Pull request by jonaro00**
## Description of change

image
image

What still remains:
- *Should we allow user to use the name even though it is taken? Prompt y/n for the rename?*
  - It's a bit tricky and not a killer. Using `--name` gets you around this check.

Bonus:
image

## How has this been tested? (if applicable)

All 3 scenarios tested locally
Updated tests

### CODE REVIEW

**Review by chesedo — APPROVED** [review]
LGTM

File: cargo-shuttle/src/lib.rs [hunk 1]
```diff
@@ -392,9 +416,21 @@ impl Shuttle {
...
+                        .expect("to have a project name provided")
+                ))
                 .default(true)
                 .interact()?;
+            if !should_create {
+                println!(
+                    "Note: The project name will not be claimed until \
+                    you start the project with `cargo shuttle project start`."
+                )
+            }
```
Comment (oddgrd) [inline cargo-shuttle/src/lib.rs:433]: Will it be clear by this point that the name is globally unique?

**Review by oddgrd — APPROVED** [review]
LGTM! Just one nit/question, but feel free to merge if you think it's clear as is.

File: cargo-shuttle/src/lib.rs (hunk 1 above)
Comment (jonaro00) [reply cargo-shuttle/src/lib.rs:433]: The prompt at name selection does tell you that. The note here is to clarify for those that do not claim the name. The word "claim" also helps.

**Pull request merged**